filesystems/stargz/fs.go

```diff
@@ -608,13 +609,19 @@ func (w *whiteout) GetAttr(out *fuse.Attr, file nodefs.File, context *fuse.Conte
 }
 
 // newState provides new state directory node.
-// It creates errFile at the same time to give it stable inode number.
-func newState(id string) *state {
+// It creates statFile at the same time to give it stable inode number.
+func newState(digest string, ur *urlReaderAt) *state {
 	return &state{
 		Node: nodefs.NewDefaultNode(),
-		id:   id,
-		err: &errFile{
+		ur:   ur,
+		statFile: &statFile{
 			Node: nodefs.NewDefaultNode(),
+			name: digest + ".json",
```

Comment:
I think the file name should be rather like `"layer-%d.json"`, but it will require adding more labels. So let's call it a day with this file name.